plugin.py

```diff
@@ -175,4 +175,8 @@ def _on_get_completions(
         if not completions:
             return
 
-        Completion(view).show(region, completions)
+        preprocess_completions(view, completions)
+
+        set_copilot_view_setting(view, "completions", completions)
+        set_copilot_view_setting(view, "completion_index", 0)
```

Comment:
I think we should handle this in the `ViewCompletionManager` as it looks more like a completion concern.

For example, we could have `completions` and `index` params in the `show` method but make them `None` by default. When they are passed then we update the view settings before getting them.

Because as of now it is a bit confusing, in order to make the `ViewCompletionManager` work you need to set something outside of it so we wouldn't be able to test this class properly without know all this prerequirements

Reply:
Like [d5679c3](https://github.com/TheSecEng/LSP-copilot/pull/13/commits/d5679c30646608e10f5c93d44b38b76a77b0d400) ?